### PR TITLE
[API] Extract user logic rules for orders to separate extension

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | master |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | Continues #13962 |
| License         | MIT                                                          |

### CODE REVIEW

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderShopUserItemExtension.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        if (!is_a($resourceClass, OrderInterface::class, true)) {
+            return;
+        }
+
+        $rootAlias = $queryBuilder->getRootAliases()[0];
+        $user = $this->userContext->getUser();
+
+        if (!$user instanceof ShopUserInterface) {
+            return;
+        }
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderShopUserItemExtension.php:49]:
```suggestion
        $user = $this->userContext->getUser();
        if (!$user instanceof ShopUserInterface) {
            return;
        }

        $rootAlias = $queryBuilder->getRootAliases()[0];
```

File: src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderVisitorItemExtension.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            return;
+        }
+
+        $rootAlias = $queryBuilder->getRootAliases()[0];
+
+        $user = $this->userContext->getUser();
+
+        if ($user !== null) {
+            return;
+        }
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderVisitorItemExtension.php:49]:
```suggestion
        $user = $this->userContext->getUser();
        if ($user !== null) {
            return;
        }
        
        $rootAlias = $queryBuilder->getRootAliases()[0];
```

File: src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderShopUserItemExtensionSpec.php [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            $queryBuilder,
+            $queryNameGenerator,
+            OrderInterface::class,
+            ['tokenValue' => 'xaza-tt_fee'],
+            Request::METHOD_PUT,
+            [ContextKeys::HTTP_REQUEST_METHOD_TYPE => Request::METHOD_PUT],
+        );
+    }
+
+    function it_does_nothing_if_object_passed_is_different_then_order(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderShopUserItemExtensionSpec.php:176]:
```suggestion
    function it_does_nothing_if_object_passed_is_different_than_order(
```

File: src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderVisitorItemExtensionSpec.php [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+use Symfony\Component\HttpFoundation\Request;
+
+final class OrderVisitorItemExtensionSpec extends ObjectBehavior
+{
+    function let(UserContextInterface $userContext): void
+    {
+        $this->beConstructedWith($userContext);
+    }
+
+    function it_filters_carts_for_visitors_to_not_authorized_for_methods_other_then_get(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderVisitorItemExtensionSpec.php:34]:
```suggestion
    function it_filters_carts_for_visitors_to_not_authorized_for_methods_other_than_get(
```

File: src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderShopUserItemExtensionSpec.php [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+use Symfony\Component\HttpFoundation\Request;
+
+final class OrderShopUserItemExtensionSpec extends ObjectBehavior
+{
+    function let(UserContextInterface $userContext): void
+    {
+        $this->beConstructedWith($userContext);
+    }
+
+    function it_filters_carts_for_shop_users_to_the_one_owned_by_them_for_methods_other_then_get(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderShopUserItemExtensionSpec.php:35]:
```suggestion
    function it_filters_carts_for_shop_users_to_the_one_owned_by_them_for_methods_other_than_get(
```

File: src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderVisitorItemExtensionSpec.php [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+            $queryBuilder,
+            $queryNameGenerator,
+            OrderInterface::class,
+            ['tokenValue' => 'xaza-tt_fee'],
+            Request::METHOD_PUT,
+            [ContextKeys::HTTP_REQUEST_METHOD_TYPE => Request::METHOD_PUT],
+        );
+    }
+
+    function it_does_nothing_if_object_passed_is_different_then_order(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/Doctrine/QueryItemExtension/OrderVisitorItemExtensionSpec.php:232]:
```suggestion
    function it_does_nothing_if_object_passed_is_different_than_order(
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Łukasz! :tada:

**Pull request merged**